src/pages/Agenda/Agenda.tsx

```diff
@@ -27,86 +36,111 @@ const Agenda: React.FC = () => {
     ));
   };
 
+  function filterSessionsByDay() {
+    const agendaItemsByDayArray = new Array<JSX.Element[]>(
+      Array.from(DAYS_MAP.keys()).length
+    );
+    for (let i = 0; i < agendaItemsByDayArray.length; i++) {
+      agendaItemsByDayArray[i] = new Array<JSX.Element>();
+    }
+    for (let i = 0; i < sessions.length; i++) {
+      const slotStart = sessions[i].slot_start;
+      if (slotStart) {
+        const day = new Date(slotStart).toDateString();
+        let dayIndex = -1;
+        switch (day) {
+          case DAYS_MAP.get("Day 1"):
+            dayIndex = 0;
+            break;
+          case DAYS_MAP.get("Day 2"):
+            dayIndex = 1;
+            break;
+          case DAYS_MAP.get("Day 3"):
+            dayIndex = 2;
+            break;
+          case DAYS_MAP.get("Day 4"):
+            dayIndex = 3;
+            break;
+          default:
+            console.log("unkown day: ", day);
+            break;
+        }
+        const sessionItem = sessions[i];
+        const shortTitle = shortenTitle(sessionItem.title);
+        const randomBoolean = Math.random() >= 0.5;
+        if (dayIndex !== -1) {
+          agendaItemsByDayArray[dayIndex].push(
+            <AgendaItem
+              key={sessionItem.id}
+              title={shortTitle}
+              startDate={dateToTime(sessionItem.slot_start)}
+              endDate={dateToTime(sessionItem.slot_end)}
+              category={sessionItem.track}
+              roomId={sessionItem.slot_roomId}
+              hearted={randomBoolean}
+            />
+          );
+        }
+      }
+    }
+
+    setAgendaItemsByDay(agendaItemsByDayArray);
+  }
+
+  useEffect(() => {
+    if (sessions.length != 0) {
+      filterSessionsByDay();
+    }
+  }, [sessions]);
+
+  useEffect(() => {
+    if (agendaItemsByDay.length > 0) {
+      const activeItemsArray: JSX.Element[] = [];
+      for (let i = 0; i < agendaItemsByDay[activeDayTab].length; i++) {
+        const items = agendaItemsByDay[activeDayTab];
+        const categoryFilter = categoryIndex
+          ? items[i].props.category === CATEGORY_FILTERS[categoryIndex]
+          : true;
+        const isYourAgenda =
+          activeAgendaTab === 1 ? items[i].props.hearted === true : true;
+
+        if (
+          items[i].props.roomId ===
+            Array.from(STAGES_MAP.keys())[activeStageTab] &&
+          isYourAgenda &&
+          categoryFilter
+        ) {
+          activeItemsArray.push(items[i]);
+        }
+      }
+      setActiveAgendaItems(activeItemsArray);
+    }
+  }, [
+    agendaItemsByDay,
+    activeDayTab,
+    activeStageTab,
+    activeAgendaTab,
+    categoryIndex,
+  ]);
```

Comment:
You shouldn't build the JSX again when you filter.
The data should be in the state and let React rerender the filtered data.

Reply:
fixed